pennylane/measurements/state.py

```diff
@@ -133,6 +133,48 @@ def return_type(self):
     def numeric_type(self):
         return complex
 
+    def shape(self, device=None):
+        if qml.active_return():
+            return self._shape_new(device)
+        if device is None:
+            raise MeasurementShapeError(
+                "The device argument is required to obtain the shape of the measurement process; "
+                + f"got return type {self.return_type}."
+            )
```

Comment:
This error only applies to state vector measurements, not density matrices. We don't need a device if we pass wires to the measurement.

Reply:
You are right, when we have `self.wires` defined we should return `(1, dim, dim)`.

However we still want to raise an error to have the same functionality as in master:
https://github.com/PennyLaneAI/pennylane/blob/62c8f40ec4f00175b160c1544dba3d00925b05c1/pennylane/measurements/measurements.py#L261

Reply:
I moved the `raise MeasurementShapeError` further down